module/skill/QuinticWalk/src/QuinticWalk.cpp

```diff
@@ -0,0 +1,383 @@
+        config.params.trunk_swing                   = cfg["walk"]["trunk"]["swing"].as<float>();
+        config.params.trunk_pause                   = cfg["walk"]["trunk"]["pause"].as<float>();
+        config.params.trunk_x_offset_p_coef_forward = cfg["walk"]["trunk"]["x_offset_p_coef"]["forward"].as<float>();
+        config.params.trunk_x_offset_p_coef_turn    = cfg["walk"]["trunk"]["x_offset_p_coef"]["turn"].as<float>();
+        config.params.trunk_pitch_p_coef_forward =
+            1.0f + cfg["walk"]["trunk"]["pitch_p_coef"]["forward"].as<Expression>();
+        config.params.trunk_pitch_p_coef_turn = 1.0f + cfg["walk"]["trunk"]["pitch_p_coef"]["turn"].as<Expression>();
+        config.params.kick_length             = cfg["walk"]["kick"]["length"].as<float>();
+        config.params.kick_phase              = cfg["walk"]["kick"]["phase"].as<float>();
+        config.params.kick_vel                = cfg["walk"]["kick"]["vel"].as<float>();
+        config.params.pause_duration          = cfg["walk"]["pause"]["duration"].as<float>();
+
+        config.max_step.x() = cfg["max_step"]["x"].as<float>();
+        config.max_step.y() = cfg["max_step"]["y"].as<float>();
+        config.max_step.z() = cfg["max_step"]["z"].as<float>();
+        config.max_step_xy  = cfg["max_step"]["xy"].as<float>();
+
+        config.imu_active          = cfg["imu"]["active"].as<bool>();
+        config.imu_pitch_threshold = 1.0f + cfg["imu"]["pitch"]["threshold"].as<float>();
+        config.imu_roll_threshold  = cfg["imu"]["roll"]["threshold"].as<float>();
+
+        for (int id = 0; id < ServoID::NUMBER_OF_SERVOS; ++id) {
+            // Sets the leg gains
+            if ((id >= ServoID::R_HIP_YAW) && (id < ServoID::HEAD_YAW)) {
+                config.jointGains[id] = cfg["gains"]["legs"].as<float>();
+            }
+            // Sets the arm gains
+            if (id < ServoID::R_HIP_YAW) {
+                config.jointGains[id] = cfg["gains"]["arms"].as<float>();
+            }
+        }
+
+        config.arm_positions.emplace_back(ServoID::R_SHOULDER_PITCH, cfg["arms"]["right_shoulder_pitch"].as<float>());
+        config.arm_positions.emplace_back(ServoID::L_SHOULDER_PITCH, cfg["arms"]["left_shoulder_pitch"].as<float>());
+        config.arm_positions.emplace_back(ServoID::R_SHOULDER_ROLL, cfg["arms"]["right_shoulder_roll"].as<float>());
+        config.arm_positions.emplace_back(ServoID::L_SHOULDER_ROLL, cfg["arms"]["left_shoulder_roll"].as<float>());
+        config.arm_positions.emplace_back(ServoID::R_ELBOW, cfg["arms"]["right_elbow"].as<float>());
+        config.arm_positions.emplace_back(ServoID::L_ELBOW, cfg["arms"]["left_elbow"].as<float>());
+    }
+
+    QuinticWalk::QuinticWalk(std::unique_ptr<NUClear::Environment> environment)
+        : BehaviourReactor(std::move(environment)) {
+
+        imu_reaction = on<Trigger<Sensors>>().then([this](const Sensors& sensors) {
```

Comment:
Shouldn't need this reaction anymore. The providers should just have a Trigger<Sensors> on them too so they can do this each update.

Reply:
This reaction doesn't run normally. Whether or not we use it is set in config. But it probably should still be under the provider since we don't really want to run it if the walk isn't running. It could be in an if block in the provider.

Reply:
Wait no it's probably better to keep it and turn it off explicitly in the Stop<Walk> and then check in Start<Walk>